checks/apps/arbor/arbor-dev.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2016-2022 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+import reframe.utility.udeps as udeps
+import uenv
+
+arbor_references = {
+    'gh200': {
+        'serial': {
+            'small': {
+                'time_run':  (9.25, -0.1, 0.1, 's'),
+            },
+            'medium': {
+                'time_run':  (35.0, -0.05, 0.05, 's'),
+            }
+        },
+        'distributed': {
+            'medium': {
+                'time_run':  (9.2, -0.05, 0.05, 's'),
+            }
+        },
+    }
+}
+
+class arbor_download(rfm.RunOnlyRegressionTest):
+    version = variable(str, value='0.9.0')
+    descr = 'Fetch Arbor sources code'
+    executable = 'wget'
+    executable_opts = [
+        f'https://github.com/arbor-sim/arbor/archive/refs/tags/v{version}.tar.gz'
+    ]
+    local = True
+
+    @sanity_function
+    def validate_download(self):
+        return sn.assert_eq(self.job.exitcode, 0)
+
+class arbor_build(rfm.CompileOnlyRegressionTest):
+    descr = 'Build Arbor'
+    valid_systems = ['*']
+    valid_prog_environs = ['+arbor-dev']
+    build_system = 'CMake'
+    maintainers = ['bcumming']
+    arbor_sources = fixture(arbor_download, scope='session')
+    # NOTE: required so that the build stage is performed on
+    # a compute node using an sbatch job.
+    # This will force the uenv and view to be loaded using
+    # "#SBATCH --uenv=" etc
+    build_locally = False
+
+    @run_before('compile')
+    def prepare_build(self):
+        self.uarch = uenv.uarch(self.current_partition)
+        self.build_system.builddir = os.path.join(self.stagedir, 'build')
+        tarball = f'v{self.arbor_sources.version}.tar.gz'
+
+        tarsource = os.path.join(self.arbor_sources.stagedir, tarball)
+        self.prebuild_cmds = [
+            f'tar --strip-components=1 -xzf{tarsource} -C {self.stagedir}'
+        ]
+
+        self.build_system.config_opts = [
+            '-DARB_WITH_MPI=on',
+            '-DARB_WITH_PYTHON=on',
+        ]
+        # set architecture-specific flags
+        if self.uarch == 'gh200':
+            self.build_system.config_opts += [
+                '-DCMAKE_CUDA_ARCHITECTURES=90',
+                '-DARB_GPU=cuda',
+            ]
+        elif self.uarch == 'a100':
+            self.build_system.config_opts += [
+                '-DCMAKE_CUDA_ARCHITECTURES=80',
+                '-DARB_GPU=cuda',
+                '-DARB_VECTORIZE=on'
+            ]
+        elif self.uarch == 'zen2':
+            self.build_system.config_opts += ['-DARB_VECTORIZE=on']
+
+        self.build_system.max_concurrency = 64
+
+        self.build_system.make_opts = ['pyarb', 'examples', 'unit']
+
+@rfm.simple_test
+class arbor_unit(rfm.RunOnlyRegressionTest):
```

Comment:
While the test is passing, I am also getting:
```
core_nid007105_232147
core_nid007105_232148
core_nid007105_232150
```

Reply:
I have no idea where they come from.
I don't think I see it on Daint - something unhealthy with Todi?